chain/chain/src/chain.rs

```diff
@@ -1452,8 +1452,11 @@ impl Chain {
             root_proofs.push(root_proofs_cur);
         }
 
-        let state_root_node =
-            self.runtime_adapter.get_state_root_node(shard_id, &chunk_header.prev_state_root())?;
+        let state_root_node = self.runtime_adapter.get_state_root_node(
+            shard_id,
+            &sync_hash,
```

Comment:
I am afraid that `sync_hash` is not what you want here because `sync_hash` is the first block of the new block and I believe here the chunk could be included in a block in the previous epoch

Reply:
Yeah, but we will disable state sync for epoch changes where resharding happens. I will add a comment here noting that.

Reply:
I created an issue https://github.com/near/nearcore/issues/4711 noting this